modules/apps/transfer/keeper/genesis.go

```diff
@@ -28,13 +28,18 @@ func (k Keeper) InitGenesis(ctx sdk.Context, state types.GenesisState) {
 	}
 
 	k.SetParams(ctx, state.Params)
+
+	for _, denomEscrow := range state.DenomEscrows {
+		k.SetTotalEscrowForDenom(ctx, denomEscrow.Denom, denomEscrow.Amount)
+	}
```

Comment:
Can we add a comment that the `DenomEscrows` must be valid as duplicate entries will cause invalid state?

Reply:
I was hoping that we could avoid ending up in invalid state if [`Validate` of `GenesisState`](https://github.com/stackman27/ibc-go/blob/stack/total-ibc-out/modules/apps/transfer/types/genesis.go#L31) is called before `InitGenesis`, and that would also call `Validate` on the collection of coins, which should error if there are duplicates. But if that is not the case, then yes, I agree we can add a comment here.

Reply:
Only a valid genesis state should be provided to `InitGenesis`. I would like a comment to be added regardless so that a future reader can understand the correctness of the code. Since `InitGenesis` is implicitly relying on a check in `genesisState.Validate()`, I think we should make a note of this, maybe in both locations